### DISCUSSION TITLE
How to find out cause of very high '(program)' time in Chrome DevTools Performance profiler?

### DISCUSSION THREAD

**Discussion by josephrocca** [denoland/deno#25796] [Q&A] [answered]
When debugging Deno programs via Chrome DevTools I always see a high percentage of '(program)':

![image](https://github.com/user-attachments/assets/a8ac6edb-a2cf-48a4-8b83-9a7de32df0ef)

![image](https://github.com/user-attachments/assets/d30c90ba-5530-4b39-8d96-af612800ec60)

![image](https://github.com/user-attachments/assets/891444c4-0538-4c20-b143-4a79adf3e299)

The hierarchical 'self' times don't add up, so I have no way of knowing what's eating up so much processing time:

![image](https://github.com/user-attachments/assets/a7b04df0-0f90-40d0-83f6-ba8b5ef0f3be)

Is this just a "me" thing? If not, I'm wondering how others debug performance bottlenecks with data like this? My understanding is that '(program)' means V8-internal, or Rust-internal. Are there perhaps debug flags I can add which would give DevTools more info?

**Comment by bartlomieju** [comment, marked as answer]
This is actually an upstream issue - see https://github.com/denoland/deno/issues/21620 for the exact same problem. Indeed `(program)` here means that we're in native code - usually it's some async code waiting to complete (eg. reading or writing from a socket), the "sync" rust code can be seen by looking for `op_` entries.

**Reply by josephrocca** [reply]
Ah, thanks!
